<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Supplement to: “HaDeX: Analysis and Visualisation of Hydrogen/Deuterium Exchange Mass Spectrometry Data” • HaDeX</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Supplement to: “HaDeX: Analysis and Visualisation of Hydrogen/Deuterium Exchange Mass Spectrometry Data”">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HaDeX</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/overview.html">Supplement to: "HaDeX: Analysis and Visualisation of Hydrogen/Deuterium Exchange Mass Spectrometry Data"</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="overview_files/htmlwidgets-1.3/htmlwidgets.js"></script><script src="overview_files/jquery-1.12.4/jquery.min.js"></script><link href="overview_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="overview_files/datatables-binding-0.5/datatables.js"></script><link href="overview_files/dt-core-bootstrap-1.10.16/css/dataTables.bootstrap.min.css" rel="stylesheet">
<link href="overview_files/dt-core-bootstrap-1.10.16/css/dataTables.bootstrap.extra.css" rel="stylesheet">
<script src="overview_files/dt-core-bootstrap-1.10.16/js/jquery.dataTables.min.js"></script><script src="overview_files/dt-core-bootstrap-1.10.16/js/dataTables.bootstrap.min.js"></script><link href="overview_files/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet">
<script src="overview_files/crosstalk-1.0.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Supplement to: “HaDeX: Analysis and Visualisation of Hydrogen/Deuterium Exchange Mass Spectrometry Data”</h1>
                        <h4 class="author">Weronika Puchała, Michał Burdukiewicz, Michał Kistowski, Katarzyna A. Dąbrowska, Aleksandra E. Badaczewska-Dawid, Dominik Cysewski, Michał Dadlez</h4>
            
            <h4 class="date">20.06.2019</h4>
      
      
      <div class="hidden name"><code>overview.Rmd</code></div>

    </div>

    
    
<div id="about-hadex" class="section level1">
<h1 class="hasAnchor">
<a href="#about-hadex" class="anchor"></a>About HaDeX</h1>
<p>HaDeX is a novel tool for processing, analysis and visualisation of HDX-MS experiments. HaDeX covers the final parts of the analytic process, including comparison of experiments, quality control and generation of publication-quality figures. To make the HaDeX <strong>R</strong> package available to the less <strong>R</strong>-fluent users, we enhanced it with a comprehensive Graphical User Interface available as a HaDeX Shiny app. The reproducibility of the whole procedure is ensured with advanced reporting functions. The app is availble online: <a href="http://mslab-ibb.pl/shiny/HaDeX/" class="uri">http://mslab-ibb.pl/shiny/HaDeX/</a> or locally through the HaDeX_gui() fuction.</p>
<p>This document covers the main functonalities of both the <strong>R</strong> package and the web server.</p>
<div id="comparison-of-the-existing-hdx-ms-software" class="section level2">
<h2 class="hasAnchor">
<a href="#comparison-of-the-existing-hdx-ms-software" class="anchor"></a>Comparison of the existing HDX-MS software</h2>
<p>To show the novelty of HaDeX, we compare its functionalities with other relatively new software for analysis of HDX-MS data: MEMHDX <span class="citation">(Hourdel et al. 2016)</span> and Deuteros <span class="citation">(Lau et al. 2019)</span>.</p>
<div id="htmlwidget-b3a5b9ae0c66ccaa21d3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b3a5b9ae0c66ccaa21d3">{"x":{"style":"bootstrap","filter":"none","data":[["Accessibility","Accessibility","Accessibility","Analysis","Analysis","Analysis","Analysis","Analysis","Visualization","Visualization","Visualization","Visualization","Visualization","Visualization","Visualization","Reporting","Reporting","Reporting","Reporting"],["Web server","Programmatic access","Desktop software","Multi-state analysis","ISO-based uncertainty","Coverage and peptide overlap","Quality control","N- and C-terminal length corrections","Global visualization of deuterium uptake","Woods plot","Zooming of the Woods plot","Customizable label names and colors","Peptide kinetics chart","3D structure visualization","Downloadable charts","Deuterium uptake download","Downloadable results of intermediate computations","Report generation","PyMol export"],["Yes","No","No","No","No","No","No","No","Yes","No","No","No","Yes","Yes","Yes","Yes","No","No","No"],["No","No","Yes","No","No","No","No","Yes","Yes","Yes","Yes","No","No","No","Yes","Yes","No","No","Yes"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","No","No","Yes","Yes","Yes","Yes","No"]],"container":"<table class=\"table table-striped table-hover\">\n  <thead>\n    <tr>\n      <th>Type<\/th>\n      <th>Name<\/th>\n      <th>MEMHDX<\/th>\n      <th>Deuteros<\/th>\n      <th>HaDeX<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","ordering":false,"paging":false,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'background-color':value == 'Yes' ? '#00BFFF' : value == 'No' ? '#FF8C91' : ''});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'background-color':value == 'Yes' ? '#00BFFF' : value == 'No' ? '#FF8C91' : ''});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'background-color':value == 'Yes' ? '#00BFFF' : value == 'No' ? '#FF8C91' : ''});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><p>We have not considered HDX Workbench <span class="citation">(Pascal et al. 2012)</span> as it deals with the preliminary steps of the analysis.</p>
</div>
</div>
<div id="hadex-functionalities" class="section level1">
<h1 class="hasAnchor">
<a href="#hadex-functionalities" class="anchor"></a>HaDeX functionalities</h1>
<div id="data-import" class="section level2">
<h2 class="hasAnchor">
<a href="#data-import" class="anchor"></a>Data import</h2>
<p>The HaDeX web server works only on data in the DynamX datafile format (Waters Corp.). The data from other sources may be also adjusted to the format accepted by HaDeX provided it has following columns:</p>
<div id="htmlwidget-f7234ac859da61089833" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f7234ac859da61089833">{"x":{"style":"bootstrap","filter":"none","data":[["Protein","Start","End","Sequence","Modification","Fragment","MaxUptake","MHP","State","Exposure","File","z","RT","Inten","Center"],["Character","Integer","Integer","Character","Logic","Logic","Numeric","Numeric","Character","Numeric","Character","Integer","Numeric","Numeric","Numeric"]],"container":"<table class=\"table table-striped table-hover\">\n  <thead>\n    <tr>\n      <th>Column.Name<\/th>\n      <th>Column.Type<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","ordering":false,"paging":false,"autoWidth":true,"order":[],"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>Although data can be imported into R using other tools, we strongly advice to rely on the <em>read_hdx()</em> function:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/read_hdx.html">read_hdx</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">"HaDeX"</span>, </a>
<a class="sourceLine" id="cb1-2" data-line-number="2">                            <span class="st">"HaDeX/data/KD_190304_Nucb2_EDTA_CaCl2_test02_clusterdata.csv"</span>))</a></code></pre></div>
<p>Currently, <em>read_hdx()</em> supports .csv, .tsv and .xls files fullfilling the data structure described above.</p>
</div>
<div id="computation-of-deuteration-levels" class="section level2">
<h2 class="hasAnchor">
<a href="#computation-of-deuteration-levels" class="anchor"></a>Computation of deuteration levels</h2>
<p>The computation of the level of deuteration involves several pre-processing steps, all of which are described in this section. These steps are performed automatically in the GUI or by the prepare_dataset() function in the console. The result of using prepare_dataset() on HDX-MS data further referred to as calc_dat.</p>
<div id="measured-data-into-overall-peptide-mass" class="section level3">
<h3 class="hasAnchor">
<a href="#measured-data-into-overall-peptide-mass" class="anchor"></a>Measured data into overall peptide mass</h3>
<p>The results of HDX-MS measurements as given in the DynamX datafiles are represented as the measured mass of peptides plus proton mass to charge ratio (<span class="math inline">\(Center\)</span>). For later use, this value has to be transformed into overall peptide mass, as shown in equation 1:</p>
<p><span class="math display">\[expMass = z \times (Center-protonMass)\tag{1}\]</span> where:</p>
<ul>
<li><p><span class="math inline">\(expMass\)</span> - result calculated from partial values,</p></li>
<li><p><span class="math inline">\(z\)</span> - charge,</p></li>
<li><p><span class="math inline">\(Center\)</span> - experimentally measured value - mass plus proton mass per charge ratio.</p></li>
</ul>
<p>HDX-MS experiments are often repeated (as the rule of thumb at least three times). Thus, we aggregate the results of replicates as the weighted mean into a single result per peptide using equation 2:</p>
<p><span class="math display">\[expMass = \frac{Inten \times expMass}{N}\tag{2}\]</span> where:</p>
<ul>
<li><p><span class="math inline">\(Inten\)</span> - intensivity,</p></li>
<li><p><span class="math inline">\(N\)</span> - number of replicates.</p></li>
</ul>
<p>The input files always encompass results of more than one measurement. The mean mass for each time point of the measurement in each state is treated as an individual result. This means that the uncertainty is calculated using the standard deviation of the mean:</p>
<p><span class="math display">\[u(x) = \sqrt{\frac{ \sum_{i=1}^n \left( x_{i} - \overline{x} \right)^2}{n(n-1)}}\tag{3}\]</span> where:</p>
<ul>
<li><p><span class="math inline">\(x_{i}\)</span> - measurement,</p></li>
<li><p><span class="math inline">\(\overline{x}\)</span> - mean value,</p></li>
<li><p><span class="math inline">\(n\)</span> - number of measurements.</p></li>
</ul>
<p>After obtaining the mass of the peptide, we can compute experimental or theoretical deuteration level depending on the chosen maximum deuteration level.</p>
</div>
<div id="experimental-deuteration-level" class="section level3">
<h3 class="hasAnchor">
<a href="#experimental-deuteration-level" class="anchor"></a>Experimental deuteration level</h3>
<p>The experimental deuteration level is computed as the deuteration level of the peptide in a specific time point <span class="math inline">\(t\)</span> compared to the deuteration level measured at the last timepoint <span class="math inline">\(out\)</span> (as as shown in equation 4). It is a function of three variables and yields a value for the chosen state and chosen time <span class="math inline">\(t\)</span>.</p>
<p><span class="math display">\[D = \frac{D_{t} - D_{0}}{D_{out} - D_{0}}\tag{4}\]</span></p>
<p>where:</p>
<ul>
<li><p><span class="math inline">\(D\)</span> - deuteration level (relative value),</p></li>
<li><p><span class="math inline">\(D_{0}\)</span> - deuteration at the beginning of the experiment (0 or close to 0, measured),</p></li>
<li><p><span class="math inline">\(D_{out}\)</span> - deuteration in <span class="math inline">\(out\)</span> tim,</p></li>
<li><p><span class="math inline">\(D_{t}\)</span> - deuteration measured in chosen time - mean value obtainted as explained before</p></li>
</ul>
<p>The equation 4 produces only relative deuteration levels. The computations of absolute deuteration levels follows a similar logic and requires only adjusting the equation 4 to its variance as shown in equation 4a:</p>
<p><span class="math display">\[D = D_{t} - D_{0}\tag{4a}\]</span> All functions in the HaDeX package contain the logical parameter <span class="math inline">\(relative\)</span> to determine if they should return absolute or relative deuteration levels.</p>
<div id="uncertainty-calculations" class="section level5">
<h5 class="hasAnchor">
<a href="#uncertainty-calculations" class="anchor"></a>Uncertainty calculations</h5>
<p>The uncertainty of a measurement is variability associated with the precision of measuring instrumentation. We present here the first derivation of uncertainty formulas for HDX-MS data according to the ISO guidelines <span class="citation">(Joint Committee for Guides in Metrology 2008)</span>. We describe the methodology of the uncertainty calculations for relative deuteration levels. The uncertainty for absolute deuteration levels is computed similarly, but without scaling.</p>
<p>To calculate uncertainty related to functions of more than one variables (e. g., equation 4) the Law of propagation of uncertainty is defined by equation 5:</p>
<p><span class="math display">\[u_{c}(y) = \sqrt{\sum_{k} \left[ \frac{\partial y}{\partial x_{k}} u(x_{k}) \right]^2}\tag{5}\]</span></p>
<p>We assume uncorrelatedness of HDX-MS measurements as they come from differently incubated samples.</p>
<p>After adapting the equation 5 using formula from equation 4, uncertainty of equation 4 is calculated as follows:</p>
<p><span class="math display">\[u_{c}(D) = \sqrt{\sum_{k} \left[ \frac{\partial D}{\partial D_{k}} u(D_{k}) \right]^2 }\tag{6}\]</span></p>
<p>where:</p>
<ul>
<li><p><span class="math inline">\(k \in {0, t, 100}\)</span>,</p></li>
<li><p><span class="math inline">\(D_{k}\)</span> - deuteration in <span class="math inline">\(k\)</span> time,</p></li>
<li><p><span class="math inline">\(u(D_{k})\)</span> - an uncertainty associated with <span class="math inline">\(D_{k}\)</span> as standard deviation of the mean value,</p></li>
</ul>
<p>Then, expanding the equation 6:</p>
<p><span class="math display">\[u_{c}(D) = \sqrt{ \left[ \frac{1}{D_{100}-D_{0}} u(D_{t}) \right]^2 + \left[ \frac{D_{t} - D_{100}}{(D_{100}-D_{0})^2} u(D_{0}) \right]^2 + \left[ \frac{D_{0} - D_{t}}{(D_{100}-D_{0})^2} u(D_{100}) \right]^2}\tag{7}\]</span> As suspected, the uncertainty associated with <span class="math inline">\(D_{t}\)</span> has the biggest impact on <span class="math inline">\(u_{c}(D)\)</span>.</p>
</div>
</div>
<div id="theoretical-deuteration-level" class="section level3">
<h3 class="hasAnchor">
<a href="#theoretical-deuteration-level" class="anchor"></a>Theoretical deuteration level</h3>
<p>In the oppostition to the experimental deuteration levels, theoetical computations only partially depend on the experimental data. Here, the maximum deuteration level comes from an imaginary peptide where all hydrogens were replaced by deuters, as it is shown in equation 8:</p>
<p><span class="math display">\[D = \frac{D_{t}-MHP}{MaxUptake \times protonMass}\tag{8}\]</span></p>
<p>where:</p>
<ul>
<li><p><span class="math inline">\(D_{t}\)</span> - deuteration measured in chosen time</p></li>
<li><p><span class="math inline">\(MHP\)</span> - theoretical mass for peptide (constant)</p></li>
<li><p><span class="math inline">\(MaxUptake\)</span> - max proton uptake for peptide (constant)</p></li>
<li><p><span class="math inline">\(protonMass\)</span> - mass of a proton (constant).</p></li>
</ul>
<p>Absolute value is calculates as in equation 8 without scalling, as shown in equation 8a:</p>
<p><span class="math display">\[D = D_{t} - MHP\tag{8a}\]</span></p>
<div id="uncertainty-calculations-1" class="section level5">
<h5 class="hasAnchor">
<a href="#uncertainty-calculations-1" class="anchor"></a>Uncertainty calculations</h5>
<p>The uncertainty of the function of one variable is defined by the equation 9:</p>
<p><span class="math display">\[u(y) = \left| \frac{dy}{dx} u(x) \right|\tag{9}\]</span></p>
<p>and based on equation 8:</p>
<p><span class="math display">\[u(D) = \left|\frac{1}{MaxUptake \times protonMass} u(D_{t}) \right|\tag{10}\]</span></p>
<p>For the absolute values, <span class="math inline">\(u(D)\)</span> is identical with <span class="math inline">\(u(D_{t})\)</span>, based on equation 8a and 9.</p>
</div>
</div>
</div>
<div id="difference-of-deuteration-levels-between-two-states" class="section level2">
<h2 class="hasAnchor">
<a href="#difference-of-deuteration-levels-between-two-states" class="anchor"></a>Difference of deuteration levels between two states</h2>
<p>The differences of deuteration levels between two states are associated with a different level of protection of hydrogens. Therefore, we are especially interested in the differential analysis of the deuteration levels. Thus, the deuteration level in one state <span class="math inline">\((D_{2})\)</span> is subtracted from deuteration level in the other state <span class="math inline">\((D_{1})\)</span>:</p>
<p><span class="math display">\[diff = D_{1} - D_{2}\tag{11}\]</span></p>
<p>and the uncertainty is a function of two variables (based on equation 11 and 5):</p>
<p><span class="math display">\[u_{c}(diff) = \sqrt{u(D_{1})^2 + u(D_{2})^2}\tag{12}\]</span></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">calc_dat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/prepare_dataset.html">prepare_dataset</a></span>(dat,</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">                            <span class="dt">in_state_first =</span> <span class="st">"gg_Nucb2_EDTA_0.001"</span>,</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">                            <span class="dt">chosen_state_first =</span> <span class="st">"gg_Nucb2_EDTA_25"</span>,</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">                            <span class="dt">out_state_first =</span> <span class="st">"gg_Nucb2_EDTA_1440"</span>,</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">                            <span class="dt">in_state_second =</span> <span class="st">"gg_Nucb2_CaCl2_0.001"</span>,</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">                            <span class="dt">chosen_state_second =</span> <span class="st">"gg_Nucb2_CaCl2_25"</span>,</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">                            <span class="dt">out_state_second =</span> <span class="st">"gg_Nucb2_CaCl2_1440"</span>) </a></code></pre></div>
</div>
<div id="visual-data-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#visual-data-analysis" class="anchor"></a>Visual data analysis</h2>
<div id="comparison-of-states" class="section level3">
<h3 class="hasAnchor">
<a href="#comparison-of-states" class="anchor"></a>Comparison of states</h3>
<p>Comparison plots show the peptide deuteration level. The x-axis represents positions of amino acids in the sequence. On the y-axis shows the level of exchange: <em>relative</em> (to the maximum deuteration) or <em>absolute</em> (in Daltons). The chart below shows peptide deuteration after 25 minutes of the incubation along with the confidence intervals. The maximum deuteration can be also computed in two different ways: either as <em>theoretical</em> (where the maximum duteration depends on the theoretical deuteration levels) and <em>experimental</em> (where the maximum deuteration is assumed to be equal to the deuteration measured at the last timepoint).</p>
<p>– theoretical:</p>
<ul>
<li>relative values:</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat,</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">                <span class="dt">theoretical =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">                <span class="dt">relative =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">                <span class="dt">state_first =</span> <span class="st">"Nucb2 Factor 1"</span>,</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">                <span class="dt">state_second =</span> <span class="st">"Nucb2 Factor 2"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">"Theoretical fraction exchanged in state comparison in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-6-1.png" width="672"></p>
<ul>
<li>absolute values:</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat,</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">                <span class="dt">theoretical =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">                <span class="dt">relative =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">                <span class="dt">state_first =</span> <span class="st">"Nucb2 Factor 1"</span>,</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">                <span class="dt">state_second =</span> <span class="st">"Nucb2 Factor 2"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">"Theoretical fraction exchanged in state comparison in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-7-1.png" width="672"></p>
<p>– experimental:</p>
<ul>
<li>relative values:</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat,</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">                <span class="dt">theoretical =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">                <span class="dt">relative =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb5-4" data-line-number="4">                <span class="dt">state_first =</span> <span class="st">"Nucb2 Factor 1"</span>,</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">                <span class="dt">state_second =</span> <span class="st">"Nucb2 Factor 2"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">"Fraction exchanged in state comparison in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-8-1.png" width="672"></p>
<ul>
<li>absolute values:</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat,</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">                <span class="dt">theoretical =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">                <span class="dt">relative =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb6-4" data-line-number="4">                <span class="dt">state_first =</span> <span class="st">"Nucb2 Factor 1"</span>,</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">                <span class="dt">state_second =</span> <span class="st">"Nucb2 Factor 2"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">"Fraction exchanged in state comparison in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-9-1.png" width="672"></p>
</div>
</div>
<div id="woods-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#woods-plot" class="anchor"></a>Woods plot</h2>
<p>Woods plot format shows the difference between the results of two different states as described by equation 11. Similarly to the comparison plot, HaDeX provides both experimental and theoretical calculations using either relative or absolute values:</p>
<p>– theoretical:</p>
<ul>
<li>relative values:</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat,</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">           <span class="dt">theoretical =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">           <span class="dt">relative =</span> <span class="ot">TRUE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-10-1.png" width="672"></p>
<ul>
<li>absolute values:</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat,</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">           <span class="dt">theoretical =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">           <span class="dt">relative =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-11-1.png" width="672"></p>
<p>– experimental:</p>
<ul>
<li>relative values:</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat,</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">           <span class="dt">theoretical =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb9-3" data-line-number="3">           <span class="dt">relative =</span> <span class="ot">TRUE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-12-1.png" width="672"></p>
<ul>
<li>absolute values:</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat,</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">           <span class="dt">theoretical =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb10-3" data-line-number="3">           <span class="dt">relative =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-13-1.png" width="672"></p>
<div id="confidence-limit-in-woods-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#confidence-limit-in-woods-plot" class="anchor"></a>Confidence limit in Woods plot</h3>
<p>The function calculate_confidence_limit_values() calculates confidence limit values as it is described elsewhere <span class="citation">(Houde, Berkowitz, and Engen 2011)</span>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw"><a href="../reference/calculate_confidence_limit_values.html">calculate_confidence_limit_values</a></span>(<span class="dt">calc_dat =</span> calc_dat,</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">                                  <span class="dt">confidence_limit =</span> <span class="fl">0.99</span>,</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">                                  <span class="dt">theoretical =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb11-4" data-line-number="4">                                  <span class="dt">relative =</span> <span class="ot">TRUE</span>)  </a></code></pre></div>
<pre><code>## [1] -0.01619004  0.01619004</code></pre>
<p>Function add_stat_dependency() returns data extended by column describing relation of a given peptide with confidence limit.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="../reference/add_stat_dependency.html">add_stat_dependency</a></span>(calc_dat, </a>
<a class="sourceLine" id="cb13-2" data-line-number="2">                   <span class="dt">confidence_limit =</span> <span class="fl">0.98</span>, </a>
<a class="sourceLine" id="cb13-3" data-line-number="3">                   <span class="dt">theoretical =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb13-4" data-line-number="4">                   <span class="dt">relative =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## # A tibble: 108 x 29
##    Sequence Start   End Med_Sequence frac_exch_state… err_frac_exch_s…
##    &lt;chr&gt;    &lt;int&gt; &lt;int&gt;        &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;
##  1 VPIDID      17    22         19.5          NaN            NaN      
##  2 KTKVKGE…    23    44         33.5          NaN            NaN      
##  3 YYDEY       45    49         47              0.975          0.00984
##  4 YYDEYL      45    50         47.5            0.679          0.00549
##  5 YLRQVID     49    55         52              0.453          0.00309
##  6 YLRQVIDV    49    56         52.5            0.428          0.00341
##  7 YLRQVID…    49    57         53              0.417          0.00271
##  8 LRQVID      50    55         52.5            0.520          0.00660
##  9 LRQVIDV     50    56         53              0.450          0.00248
## 10 LRQVIDVL    50    57         53.5            0.414          0.00346
## # … with 98 more rows, and 23 more variables: frac_exch_state_2 &lt;dbl&gt;,
## #   err_frac_exch_state_2 &lt;dbl&gt;, diff_frac_exch &lt;dbl&gt;,
## #   err_frac_exch &lt;dbl&gt;, abs_frac_exch_state_1 &lt;dbl&gt;,
## #   err_abs_frac_exch_state_1 &lt;dbl&gt;, abs_frac_exch_state_2 &lt;dbl&gt;,
## #   err_abs_frac_exch_state_2 &lt;dbl&gt;, abs_diff_frac_exch &lt;dbl&gt;,
## #   err_abs_diff_frac_exch &lt;dbl&gt;, avg_theo_in_time_1 &lt;dbl&gt;,
## #   err_avg_theo_in_time_1 &lt;dbl&gt;, avg_theo_in_time_2 &lt;dbl&gt;,
## #   err_avg_theo_in_time_2 &lt;dbl&gt;, diff_theo_frac_exch &lt;dbl&gt;,
## #   err_diff_theo_frac_exch &lt;dbl&gt;, abs_avg_theo_in_time_1 &lt;dbl&gt;,
## #   err_abs_avg_theo_in_time_1 &lt;dbl&gt;, abs_avg_theo_in_time_2 &lt;dbl&gt;,
## #   err_abs_avg_theo_in_time_2 &lt;dbl&gt;, abs_diff_theo_frac_exch &lt;dbl&gt;,
## #   err_abs_diff_theo_frac_exch &lt;dbl&gt;, valid_at_0.98 &lt;lgl&gt;</code></pre>
</div>
</div>
<div id="additional-tools" class="section level2">
<h2 class="hasAnchor">
<a href="#additional-tools" class="anchor"></a>Additional tools</h2>
<p>HaDeX provides additional tools for assement of experiments.</p>
<div id="peptide-coverage" class="section level3">
<h3 class="hasAnchor">
<a href="#peptide-coverage" class="anchor"></a>Peptide coverage</h3>
<p>The sequence of protein(s) is reconstructed from the input file and based of provided peptides. Thus, parts of the sequences not covered by peptides are marked as X according to the IUPAC convention for unknown amino acids. The sequence is reconstructed using the reconstruct_sequence function.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="../reference/reconstruct_sequence.html">reconstruct_sequence</a></span>(dat)</a></code></pre></div>
<pre><code>## [1] "xxxxxxxxxxxxxxxxVPIDIDKTKVKGEGHVEGEKIENPDTGLYYDEYLRQVIDVLETDKHFREKLQTADIEEIKSGKLSRELDLVSHHVRTRLDELKRQEVARLRMLIKAKMDSVQDTGIDHQALLKQFEHLNHQNPDTFEPKDLDMLIKAATSDLENYDKTRHEEFKKYEMxxxxxxxxxxxxLDEEKRQREESKFGEMxxxxxxxxxxxxxxxxxxxKEVWEEADGLDPNEFDPKTFFKLHDVNNDRFLDEQELEAxFTKELEKVYDPKNEEDDMVEMEEERLxxxxHVMNEVDINKDRLVTLEEFLRATEKKEFLEPDSWETLDQQQLFTEDELKEFESHISQQEDELRKKAEELQKQKEELQRQHDQLQAQEQELQQVVKQMEQKKLQQANPPAGPAGELK"</code></pre>
<p>Additionally, the coverage of peptides can be presented on a chart using the plot_coverage() and plot_position_frequency() functions.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="../reference/plot_coverage.html">plot_coverage</a></span>(dat, <span class="dt">chosen_state =</span> <span class="st">"gg_Nucb2_CaCl2"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-17-1.png" width="672"></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw"><a href="../reference/plot_position_frequency.html">plot_position_frequency</a></span>(dat, <span class="dt">chosen_state =</span> <span class="st">"gg_Nucb2_CaCl2"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-17-2.png" width="672"></p>
<p>User can choose which state (or states) should be included in these plots. If this parameter is not provided, the first possible state is chosen. If a given peptide is available in more than one state, it is shown only once.</p>
</div>
<div id="quality-control" class="section level3">
<h3 class="hasAnchor">
<a href="#quality-control" class="anchor"></a>Quality control</h3>
<p>Function quality_control() shows how the mean deuteration uncertainty per peptide changes with time points of an experiment. Therefore, a user can detect a time point after which the decrease of the deuteration uncertainty becomes too marginal to prolong the measurements. This function is the most useful in the case of multiple measurements of the same or very similar proteins, where it allows to optimize the duration of the incubation.</p>
<p>Example:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">result &lt;-<span class="st"> </span><span class="kw"><a href="../reference/quality_control.html">quality_control</a></span>(<span class="dt">dat =</span> dat,</a>
<a class="sourceLine" id="cb19-2" data-line-number="2">                          <span class="dt">state_first =</span> <span class="st">"gg_Nucb2_EDTA"</span>,</a>
<a class="sourceLine" id="cb19-3" data-line-number="3">                          <span class="dt">state_second =</span> <span class="st">"gg_Nucb2_CaCl2"</span>, </a>
<a class="sourceLine" id="cb19-4" data-line-number="4">                          <span class="dt">chosen_time =</span> <span class="dv">25</span>, </a>
<a class="sourceLine" id="cb19-5" data-line-number="5">                          <span class="dt">in_time =</span> <span class="fl">0.001</span>, </a>
<a class="sourceLine" id="cb19-6" data-line-number="6">                          <span class="dt">relative =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>The function returns an object of the data.frame class, that can be easily visualized.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">ggplot</span>(result[result[<span class="st">"time"</span>]<span class="op">&gt;=</span><span class="dv">1</span>,]) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> time, <span class="dt">y =</span> avg_err_state_first, <span class="dt">color =</span> <span class="st">"Average error (first state)"</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb20-3" data-line-number="3"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> time, <span class="dt">y =</span> avg_err_state_second, <span class="dt">color =</span> <span class="st">"Average error (second state)"</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb20-4" data-line-number="4"><span class="st">  </span><span class="kw">scale_x_log10</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb20-5" data-line-number="5"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">"log(time) [min]"</span>, <span class="dt">y =</span> <span class="st">"Average uncertainty"</span>, <span class="dt">title =</span> <span class="st">"Uncertainty change"</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb20-6" data-line-number="6"><span class="st">  </span><span class="kw">theme_bw</span>(<span class="dt">base_size =</span> <span class="dv">11</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb20-7" data-line-number="7"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">"bottom"</span>,</a>
<a class="sourceLine" id="cb20-8" data-line-number="8">        <span class="dt">legend.title =</span> <span class="kw">element_blank</span>())</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-19-1.png" width="672"></p>
<p>Example above is based on relative values. Although HaDeX provides results in absolute values, be aware that absolute calculations do not encompass time out (it is used for scaling), so the uncertainty is not the function of <span class="math inline">\(D_{100}\)</span>, but is constant.</p>
</div>
</div>
</div>
<div id="hadex-graphical-user-interface" class="section level1">
<h1 class="hasAnchor">
<a href="#hadex-graphical-user-interface" class="anchor"></a>HaDeX Graphical User Interface</h1>
<p>The HaDeX Shiny app is launched by the HaDeX_gui() function or available at MS Lab website: <a href="http://mslab-ibb.pl/shiny/HaDeX/" class="uri">http://mslab-ibb.pl/shiny/HaDeX/</a>.</p>
</div>
<div id="examples" class="section level1">
<h1 class="hasAnchor">
<a href="#examples" class="anchor"></a>Examples</h1>
<div id="example-1-cd160-hvem" class="section level2">
<h2 class="hasAnchor">
<a href="#example-1-cd160-hvem" class="anchor"></a>Example 1: CD160-HVEM</h2>
<p>The interaction between HVEM and the <a href="https://www.uniprot.org/uniprot/O95971">CD160</a> receptor was measured with HDX-MS.</p>
<p>Firstly, we read the input data, exactly as provided by theDynamX 3.0 (Waters Corp.)</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(HaDeX)</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="co"># file import</span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4">dat_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/read_hdx.html">read_hdx</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">"HaDeX"</span>, </a>
<a class="sourceLine" id="cb21-5" data-line-number="5">                              <span class="st">"HaDeX/data/KD_180110_CD160_HVEM.csv"</span>))</a></code></pre></div>
<p>Then, we reconstruct the protein sequence using peptides measured during the experiment. We observe tha region from amino acid 107 till amino acid 124 is not covered by any peptide.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw"><a href="../reference/reconstruct_sequence.html">reconstruct_sequence</a></span>(dat_<span class="dv">1</span>)</a></code></pre></div>
<pre><code>## [1] "INITSSASQEGTRLNLICTVWHKKEEAEGFVVFLCKDRSGDCSPETSLKQLRLKRDPGIDGVGEISSQLMFTISQVTPLHSGTYQCCARSQKSGIRLQGHFFSILFxxxxxxxxxxxxxxxxxxFSHNEGTL"</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw"><a href="../reference/plot_position_frequency.html">plot_position_frequency</a></span>(dat_<span class="dv">1</span>, <span class="dt">chosen_state =</span> <span class="st">"CD160"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-21-1.png" width="672"></p>
<p>On the theoretical plot are visible regions, which exchange quickly (N-terminus, regions between 30-70 amino acid) and regions, which exchange slowly (peptides 15-24, 95-110 amino acid). We can also see the differences in exchange between two states, indicating regions which changed upon binding with other protein.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="co"># calculate data</span></a>
<a class="sourceLine" id="cb25-2" data-line-number="2">calc_dat_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/prepare_dataset.html">prepare_dataset</a></span>(<span class="dt">dat =</span> dat_<span class="dv">1</span>,</a>
<a class="sourceLine" id="cb25-3" data-line-number="3">                              <span class="dt">in_state_first =</span> <span class="st">"CD160_0.001"</span>,</a>
<a class="sourceLine" id="cb25-4" data-line-number="4">                              <span class="dt">chosen_state_first =</span> <span class="st">"CD160_1"</span>,</a>
<a class="sourceLine" id="cb25-5" data-line-number="5">                              <span class="dt">out_state_first =</span> <span class="st">"CD160_1440"</span>,</a>
<a class="sourceLine" id="cb25-6" data-line-number="6">                              <span class="dt">in_state_second =</span> <span class="st">"CD160_HVEM_0.001"</span>,</a>
<a class="sourceLine" id="cb25-7" data-line-number="7">                              <span class="dt">chosen_state_second =</span> <span class="st">"CD160_HVEM_1"</span>,</a>
<a class="sourceLine" id="cb25-8" data-line-number="8">                              <span class="dt">out_state_second =</span> <span class="st">"CD160_HVEM_1440"</span>)                             </a>
<a class="sourceLine" id="cb25-9" data-line-number="9"></a>
<a class="sourceLine" id="cb25-10" data-line-number="10"><span class="co"># theoretical comparison plot - relative values</span></a>
<a class="sourceLine" id="cb25-11" data-line-number="11"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">1</span>,</a>
<a class="sourceLine" id="cb25-12" data-line-number="12">                <span class="dt">theoretical =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb25-13" data-line-number="13">                <span class="dt">relative =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb25-14" data-line-number="14">                <span class="dt">state_first =</span> <span class="st">"CD160"</span>,</a>
<a class="sourceLine" id="cb25-15" data-line-number="15">                <span class="dt">state_second =</span> <span class="st">"CD160_HVEM"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-22-1.png" width="672"></p>
<p>On the experimental plot are visible regions, which exchange quickly (N terminal part, regions between 30-70 amino acid) and regions, which exchange slowly (peptides 15-24, 30-35, 95-110 amino acid).</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="co"># experimental comparison plot - relative values</span></a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">1</span>,</a>
<a class="sourceLine" id="cb26-3" data-line-number="3">                <span class="dt">theoretical =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb26-4" data-line-number="4">                <span class="dt">relative =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb26-5" data-line-number="5">                <span class="dt">state_first =</span> <span class="st">"CD160"</span>,</a>
<a class="sourceLine" id="cb26-6" data-line-number="6">                <span class="dt">state_second =</span> <span class="st">"CD160_HVEM"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-23-1.png" width="672"></p>
<p>Plots below are equivalent to plots above but in absolute values - for users that prefer those.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1"><span class="co"># theoretical comparison plot - absolute values</span></a>
<a class="sourceLine" id="cb27-2" data-line-number="2"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">1</span>,</a>
<a class="sourceLine" id="cb27-3" data-line-number="3">                <span class="dt">theoretical =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb27-4" data-line-number="4">                <span class="dt">relative =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb27-5" data-line-number="5">                <span class="dt">state_first =</span> <span class="st">"CD160"</span>,</a>
<a class="sourceLine" id="cb27-6" data-line-number="6">                <span class="dt">state_second =</span> <span class="st">"CD160_HVEM"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-24-1.png" width="672"></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="co"># experimental comparison plot - absolute values</span></a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">1</span>,</a>
<a class="sourceLine" id="cb28-3" data-line-number="3">                <span class="dt">theoretical =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb28-4" data-line-number="4">                <span class="dt">relative =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb28-5" data-line-number="5">                <span class="dt">state_first =</span> <span class="st">"CD160"</span>,</a>
<a class="sourceLine" id="cb28-6" data-line-number="6">                <span class="dt">state_second =</span> <span class="st">"CD160_HVEM"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-24-2.png" width="672"></p>
<p>Wood’s plot is a different way of presenting the same data as in comparison plot. In Wood’s plot the levels of peptides’ deuteration (either relative or absolute) are subtracted between states and presented with error bars. The plot below shows peptides, which levels of deuteration were significantly lower upon binding with other protein (red) and peptides, which levels of exchange were significantly higher upon binding with other protein (blue). The plot also show peptides in which no significant changes in deuteration between states are visible (grey). The biggest changes on the theoretical plot can be observed in 3 peptides (15-24, 30-35, 75-90).</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="co"># theoretical Woods plot - relative values</span></a>
<a class="sourceLine" id="cb29-2" data-line-number="2"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">1</span>,</a>
<a class="sourceLine" id="cb29-3" data-line-number="3">           <span class="dt">theoretical =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb29-4" data-line-number="4">           <span class="dt">relative =</span> <span class="ot">TRUE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb29-5" data-line-number="5"><span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="op">-</span>.<span class="dv">2</span>, <span class="fl">.2</span>))</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-25-1.png" width="672"></p>
<p>The biggest changes on the experimental plot can be observed in 3 peptides (15-24, 110-115).</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="co"># experimental Woods plot - relative values</span></a>
<a class="sourceLine" id="cb30-2" data-line-number="2"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">1</span>,</a>
<a class="sourceLine" id="cb30-3" data-line-number="3">           <span class="dt">theoretical =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb30-4" data-line-number="4">           <span class="dt">relative =</span> <span class="ot">TRUE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb30-5" data-line-number="5"><span class="st">  </span><span class="kw">coord_cartesian</span>(<span class="dt">ylim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="op">-</span>.<span class="dv">2</span>, <span class="fl">.2</span>))</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-26-1.png" width="672"></p>
<p>Plot below show results in absolute values.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="co"># theoretical Woods plot - absolute values</span></a>
<a class="sourceLine" id="cb31-2" data-line-number="2"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">1</span>,</a>
<a class="sourceLine" id="cb31-3" data-line-number="3">           <span class="dt">theoretical =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb31-4" data-line-number="4">           <span class="dt">relative =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb31-5" data-line-number="5"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">"Theoretical fraction exchanged between states in 1 min time"</span>) </a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-27-1.png" width="672"></p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1"><span class="co"># experimental Woods plot - absolute values</span></a>
<a class="sourceLine" id="cb32-2" data-line-number="2"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">1</span>,</a>
<a class="sourceLine" id="cb32-3" data-line-number="3">           <span class="dt">theoretical =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb32-4" data-line-number="4">           <span class="dt">relative =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb32-5" data-line-number="5"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">"Fraction exchanged between states in 1 min time"</span>) </a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-27-2.png" width="672"></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="co"># quality control - relative values </span></a>
<a class="sourceLine" id="cb33-2" data-line-number="2">(result &lt;-<span class="st"> </span><span class="kw"><a href="../reference/quality_control.html">quality_control</a></span>(<span class="dt">dat =</span> dat_<span class="dv">1</span>,</a>
<a class="sourceLine" id="cb33-3" data-line-number="3">                <span class="dt">state_first =</span> <span class="st">"CD160"</span>,</a>
<a class="sourceLine" id="cb33-4" data-line-number="4">                <span class="dt">state_second =</span> <span class="st">"CD160_HVEM"</span>, </a>
<a class="sourceLine" id="cb33-5" data-line-number="5">                <span class="dt">chosen_time =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb33-6" data-line-number="6">                <span class="dt">in_time =</span> <span class="fl">0.001</span>))</a></code></pre></div>
<pre><code>##       time avg_err_state_first sd_err_state_first avg_err_state_second
## 1    0.167         0.003919621        0.005116853          0.005738668
## 2    1.000         0.002607188        0.003708500          0.004930354
## 3    5.000         0.002925674        0.002207625          0.003811229
## 4   25.000         0.002086275        0.001636480          0.003375989
## 5  120.000         0.002124354        0.001284390          0.002759393
## 6 1440.000         0.001841383        0.001032972          0.002532581
##   sd_err_state_second avg_err_theo_state_first sd_err_theo_state_first
## 1         0.007625134             0.0007087017            0.0005185554
## 2         0.006697768             0.0007087017            0.0005185554
## 3         0.004561764             0.0007087017            0.0005185554
## 4         0.003690424             0.0007087017            0.0005185554
## 5         0.001976257             0.0007087017            0.0005185554
## 6         0.001120593             0.0007087017            0.0005185554
##   avg_err_theo_state_second sd_err_theo_state_second    avg_diff
## 1               0.001236524             0.0005534176 0.007045439
## 2               0.001236524             0.0005534176 0.005709252
## 3               0.001236524             0.0005534176 0.004920756
## 4               0.001236524             0.0005534176 0.004053767
## 5               0.001236524             0.0005534176 0.003470818
## 6               0.001236524             0.0005534176 0.003200211
##       sd_diff avg_theo_diff sd_theo_diff
## 1 0.009107625   0.001476534 0.0006500359
## 2 0.007555534   0.001476534 0.0006500359
## 3 0.004952401   0.001476534 0.0006500359
## 4 0.003949308   0.001476534 0.0006500359
## 5 0.002088545   0.001476534 0.0006500359
## 6 0.001369362   0.001476534 0.0006500359</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="co"># example quality control visualisation</span></a>
<a class="sourceLine" id="cb35-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)</a>
<a class="sourceLine" id="cb35-3" data-line-number="3"><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(result) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x =</span> time, <span class="dt">y =</span> avg_err_state_first, <span class="dt">color =</span> <span class="st">"Average error (first state)"</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x =</span> time, <span class="dt">y =</span> avg_err_state_second, <span class="dt">color =</span> <span class="st">"Average error (second state)"</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb35-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/lims.html">ylim</a></span>(<span class="dv">0</span>, <span class="fl">0.05</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-8" data-line-number="8"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="dt">x =</span> <span class="st">"log(time) [min]"</span>, <span class="dt">y =</span> <span class="st">"Average uncertainty"</span>, <span class="dt">title =</span> <span class="st">"Uncertainty change in out time"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-9" data-line-number="9"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">legend.title =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>(),</a>
<a class="sourceLine" id="cb35-10" data-line-number="10">        <span class="dt">legend.position =</span> <span class="st">"bottom"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-27-3.png" width="672"></p>
</div>
<div id="example-workflow-2" class="section level2">
<h2 class="hasAnchor">
<a href="#example-workflow-2" class="anchor"></a>Example workflow 2</h2>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(HaDeX)</a>
<a class="sourceLine" id="cb36-2" data-line-number="2"></a>
<a class="sourceLine" id="cb36-3" data-line-number="3"><span class="co"># file import</span></a>
<a class="sourceLine" id="cb36-4" data-line-number="4">dat_<span class="dv">2</span> &lt;-<span class="st">  </span><span class="kw"><a href="../reference/read_hdx.html">read_hdx</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">"HaDeX"</span>, </a>
<a class="sourceLine" id="cb36-5" data-line-number="5">                               <span class="st">"HaDeX/data/KD_190304_Nucb2_EDTA_CaCl2_test02_clusterdata.csv"</span>))</a>
<a class="sourceLine" id="cb36-6" data-line-number="6"></a>
<a class="sourceLine" id="cb36-7" data-line-number="7"><span class="co"># protein sequence reconstruction</span></a>
<a class="sourceLine" id="cb36-8" data-line-number="8"><span class="kw"><a href="../reference/reconstruct_sequence.html">reconstruct_sequence</a></span>(dat_<span class="dv">2</span>)</a></code></pre></div>
<pre><code>## [1] "xxxxxxxxxxxxxxxxVPIDIDKTKVKGEGHVEGEKIENPDTGLYYDEYLRQVIDVLETDKHFREKLQTADIEEIKSGKLSRELDLVSHHVRTRLDELKRQEVARLRMLIKAKMDSVQDTGIDHQALLKQFEHLNHQNPDTFEPKDLDMLIKAATSDLENYDKTRHEEFKKYEMxxxxxxxxxxxxLDEEKRQREESKFGEMxxxxxxxxxxxxxxxxxxxKEVWEEADGLDPNEFDPKTFFKLHDVNNDRFLDEQELEAxFTKELEKVYDPKNEEDDMVEMEEERLxxxxHVMNEVDINKDRLVTLEEFLRATEKKEFLEPDSWETLDQQQLFTEDELKEFESHISQQEDELRKKAEELQKQKEELQRQHDQLQAQEQELQQVVKQMEQKKLQQANPPAGPAGELK"</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1"><span class="co"># calculate data</span></a>
<a class="sourceLine" id="cb38-2" data-line-number="2">calc_dat_<span class="dv">2</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/prepare_dataset.html">prepare_dataset</a></span>(<span class="dt">dat =</span> dat_<span class="dv">2</span>,</a>
<a class="sourceLine" id="cb38-3" data-line-number="3">                              <span class="dt">in_state_first =</span> <span class="st">"gg_Nucb2_EDTA_0.001"</span>,</a>
<a class="sourceLine" id="cb38-4" data-line-number="4">                              <span class="dt">chosen_state_first =</span> <span class="st">"gg_Nucb2_EDTA_25"</span>,</a>
<a class="sourceLine" id="cb38-5" data-line-number="5">                              <span class="dt">out_state_first =</span> <span class="st">"gg_Nucb2_EDTA_1440"</span>,</a>
<a class="sourceLine" id="cb38-6" data-line-number="6">                              <span class="dt">in_state_second =</span> <span class="st">"gg_Nucb2_CaCl2_0.001"</span>,</a>
<a class="sourceLine" id="cb38-7" data-line-number="7">                              <span class="dt">chosen_state_second =</span> <span class="st">"gg_Nucb2_CaCl2_25"</span>,</a>
<a class="sourceLine" id="cb38-8" data-line-number="8">                              <span class="dt">out_state_second =</span> <span class="st">"gg_Nucb2_CaCl2_1440"</span>)                             </a>
<a class="sourceLine" id="cb38-9" data-line-number="9"></a>
<a class="sourceLine" id="cb38-10" data-line-number="10"><span class="co"># theoretical comparison plot - relative values</span></a>
<a class="sourceLine" id="cb38-11" data-line-number="11"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">2</span>,</a>
<a class="sourceLine" id="cb38-12" data-line-number="12">                <span class="dt">theoretical =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb38-13" data-line-number="13">                <span class="dt">relative =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb38-14" data-line-number="14">                <span class="dt">state_first =</span> <span class="st">"Nucb2 Factor 1"</span>,</a>
<a class="sourceLine" id="cb38-15" data-line-number="15">                <span class="dt">state_second =</span> <span class="st">"Nucb2 Factor 2"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb38-16" data-line-number="16"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="dt">title =</span> <span class="st">"Theoretical fraction exchanged in </span><span class="ch">\n</span><span class="st">state comparison in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-28-1.png" width="672"></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1"><span class="co"># experimental comparison plot - relative values </span></a>
<a class="sourceLine" id="cb39-2" data-line-number="2"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">2</span>,</a>
<a class="sourceLine" id="cb39-3" data-line-number="3">                <span class="dt">theoretical =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb39-4" data-line-number="4">                <span class="dt">relative =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb39-5" data-line-number="5">                <span class="dt">state_first =</span> <span class="st">"Nucb2 Factor 1"</span>,</a>
<a class="sourceLine" id="cb39-6" data-line-number="6">                <span class="dt">state_second =</span> <span class="st">"Nucb2 Factor 2"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="dt">title =</span> <span class="st">"Fraction exchanged in </span><span class="ch">\n</span><span class="st">state comparison in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-28-2.png" width="672"></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1"><span class="co"># theoretical comparison plot - absolute values</span></a>
<a class="sourceLine" id="cb40-2" data-line-number="2"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">2</span>,</a>
<a class="sourceLine" id="cb40-3" data-line-number="3">                <span class="dt">theoretical =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb40-4" data-line-number="4">                <span class="dt">relative =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb40-5" data-line-number="5">                <span class="dt">state_first =</span> <span class="st">"Nucb2 Factor 1"</span>,</a>
<a class="sourceLine" id="cb40-6" data-line-number="6">                <span class="dt">state_second =</span> <span class="st">"Nucb2 Factor 2"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb40-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="dt">title =</span> <span class="st">"Theoretical fraction exchanged in </span><span class="ch">\n</span><span class="st">state comparison in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-28-3.png" width="672"></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1"><span class="co"># experimental comparison plot - absolute values </span></a>
<a class="sourceLine" id="cb41-2" data-line-number="2"><span class="kw"><a href="../reference/comparison_plot.html">comparison_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">2</span>,</a>
<a class="sourceLine" id="cb41-3" data-line-number="3">                <span class="dt">theoretical =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb41-4" data-line-number="4">                <span class="dt">relative =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb41-5" data-line-number="5">                <span class="dt">state_first =</span> <span class="st">"Nucb2 Factor 1"</span>,</a>
<a class="sourceLine" id="cb41-6" data-line-number="6">                <span class="dt">state_second =</span> <span class="st">"Nucb2 Factor 2"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb41-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="dt">title =</span> <span class="st">"Fraction exchanged in </span><span class="ch">\n</span><span class="st">state comparison in 25 min time"</span>)</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-28-4.png" width="672"></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1"><span class="co"># theoretical Woods plot - relative values</span></a>
<a class="sourceLine" id="cb42-2" data-line-number="2"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">2</span>,</a>
<a class="sourceLine" id="cb42-3" data-line-number="3">           <span class="dt">theoretical =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb42-4" data-line-number="4">           <span class="dt">relative =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb42-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="dt">title =</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb42-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span>(<span class="dt">ylim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="op">-</span>.<span class="dv">5</span>, <span class="fl">.7</span>))</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-28-5.png" width="672"></p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1"><span class="co"># experimental Woods plot - relative values</span></a>
<a class="sourceLine" id="cb43-2" data-line-number="2"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">2</span>,</a>
<a class="sourceLine" id="cb43-3" data-line-number="3">           <span class="dt">theoretical =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb43-4" data-line-number="4">           <span class="dt">relative =</span> <span class="ot">TRUE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb43-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="dt">title =</span> <span class="st">"Fraction exchanged between states in 25 min time"</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb43-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span>(<span class="dt">ylim =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="op">-</span>.<span class="dv">5</span>, <span class="fl">.7</span>))</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-28-6.png" width="672"></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1"><span class="co"># theoretical Woods plot - absolute values</span></a>
<a class="sourceLine" id="cb44-2" data-line-number="2"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">2</span>,</a>
<a class="sourceLine" id="cb44-3" data-line-number="3">           <span class="dt">theoretical =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb44-4" data-line-number="4">           <span class="dt">relative =</span> <span class="ot">FALSE</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb44-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="dt">title =</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span>) </a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-28-7.png" width="672"></p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1"><span class="co"># experimental Woods plot - absolute values</span></a>
<a class="sourceLine" id="cb45-2" data-line-number="2"><span class="kw"><a href="../reference/woods_plot.html">woods_plot</a></span>(<span class="dt">calc_dat =</span> calc_dat_<span class="dv">2</span>,</a>
<a class="sourceLine" id="cb45-3" data-line-number="3">           <span class="dt">theoretical =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb45-4" data-line-number="4">           <span class="dt">relative =</span> <span class="ot">FALSE</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb45-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="dt">title =</span> <span class="st">"Fraction exchanged between states in 25 min time"</span>) </a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-28-8.png" width="672"></p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1"><span class="co"># quality control</span></a>
<a class="sourceLine" id="cb46-2" data-line-number="2">(result &lt;-<span class="st"> </span><span class="kw"><a href="../reference/quality_control.html">quality_control</a></span>(<span class="dt">dat =</span> dat_<span class="dv">2</span>,</a>
<a class="sourceLine" id="cb46-3" data-line-number="3">                           <span class="dt">state_first =</span> <span class="st">"gg_Nucb2_EDTA"</span>,</a>
<a class="sourceLine" id="cb46-4" data-line-number="4">                           <span class="dt">state_second =</span> <span class="st">"gg_Nucb2_CaCl2"</span>, </a>
<a class="sourceLine" id="cb46-5" data-line-number="5">                           <span class="dt">chosen_time =</span> <span class="dv">25</span>, </a>
<a class="sourceLine" id="cb46-6" data-line-number="6">                           <span class="dt">in_time =</span> <span class="fl">0.001</span>))</a></code></pre></div>
<pre><code>##       time avg_err_state_first sd_err_state_first avg_err_state_second
## 1    0.167         0.158083662        0.374221748          0.808330769
## 2    1.000         0.030622838        0.060927377          0.078091161
## 3   10.000         0.006651472        0.005621178          0.009344373
## 4   25.000         0.005796828        0.003283214          0.009559405
## 5   60.000         0.005530984        0.002672025          0.007094412
## 6 1440.000         0.005308684        0.002471285          0.006814869
##   sd_err_state_second avg_err_theo_state_first sd_err_theo_state_first
## 1         2.551580562              0.001396896            0.0006199273
## 2         0.146589091              0.001396896            0.0006199273
## 3         0.005126022              0.001396896            0.0006199273
## 4         0.002774018              0.001396896            0.0006199273
## 5         0.002096056              0.001396896            0.0006199273
## 6         0.002019402              0.001396896            0.0006199273
##   avg_err_theo_state_second sd_err_theo_state_second    avg_diff
## 1               0.002539802             0.0007520063 0.563316031
## 2               0.002539802             0.0007520063 0.080138397
## 3               0.002539802             0.0007520063 0.011699777
## 4               0.002539802             0.0007520063 0.011419694
## 5               0.002539802             0.0007520063 0.009074865
## 6               0.002539802             0.0007520063 0.008631836
##       sd_diff avg_theo_diff sd_theo_diff
## 1 1.248992873   0.002982639 0.0006774071
## 2 0.138961471   0.002982639 0.0006774071
## 3 0.006096753   0.002982639 0.0006774071
## 4 0.002828424   0.002982639 0.0006774071
## 5 0.002499717   0.002982639 0.0006774071
## 6 0.002732523   0.002982639 0.0006774071</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" data-line-number="1"><span class="co"># example quality control visualisation - relative values</span></a>
<a class="sourceLine" id="cb48-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)</a>
<a class="sourceLine" id="cb48-3" data-line-number="3"><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(result[result[<span class="st">"time"</span>]<span class="op">&gt;=</span><span class="dv">1</span>,]) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb48-4" data-line-number="4"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x =</span> time, <span class="dt">y =</span> avg_err_state_first, <span class="dt">color =</span> <span class="st">"Average error (first state)"</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb48-5" data-line-number="5"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x =</span> time, <span class="dt">y =</span> avg_err_state_second, <span class="dt">color =</span> <span class="st">"Average error (second state)"</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb48-6" data-line-number="6"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb48-7" data-line-number="7"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="dt">x =</span> <span class="st">"log(time) [min]"</span>, <span class="dt">y =</span> <span class="st">"Average uncertainty"</span>, <span class="dt">title =</span> <span class="st">"Uncertainty change"</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb48-8" data-line-number="8"><span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">legend.position =</span> <span class="st">"bottom"</span>,</a>
<a class="sourceLine" id="cb48-9" data-line-number="9">        <span class="dt">legend.title =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>())</a></code></pre></div>
<p><img src="overview_files/figure-html/unnamed-chunk-28-9.png" width="672"></p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-HoudeUtilityHydrogenDeuterium2011">
<p>Houde, Damian, Steven A. Berkowitz, and John R. Engen. 2011. “The Utility of Hydrogen/Deuterium Exchange Mass Spectrometry in Biopharmaceutical Comparability Studies.” <em>Journal of Pharmaceutical Sciences</em> 100 (6): 2071–86. <a href="https://doi.org/10.1002/jps.22432">https://doi.org/10.1002/jps.22432</a>.</p>
</div>
<div id="ref-HourdelMEMHDXinteractivetool2016">
<p>Hourdel, Véronique, Stevenn Volant, Darragh P. O’Brien, Alexandre Chenal, Julia Chamot-Rooke, Marie-Agnès Dillies, and Sébastien Brier. 2016. “MEMHDX: An Interactive Tool to Expedite the Statistical Validation and Visualization of Large HDX-MS Datasets.” <em>Bioinformatics</em> 32 (22): 3413–9. <a href="https://doi.org/10.1093/bioinformatics/btw420">https://doi.org/10.1093/bioinformatics/btw420</a>.</p>
</div>
<div id="ref-jcgm:2008:EMDG">
<p>Joint Committee for Guides in Metrology. 2008. “JCGM 100: Evaluation of Measurement Data - Guide to the Expression of Uncertainty in Measurement.” JCGM.</p>
</div>
<div id="ref-LauDeuterossoftwarerapid2019">
<p>Lau, Andy M. C., Zainab Ahdash, Chloe Martens, and Argyris Politis. 2019. “Deuteros: Software for Rapid Analysis and Visualization of Data from Differential Hydrogen Deuterium Exchange-Mass Spectrometry.” <em>Bioinformatics (Oxford, England)</em>, January. <a href="https://doi.org/10.1093/bioinformatics/btz022">https://doi.org/10.1093/bioinformatics/btz022</a>.</p>
</div>
<div id="ref-PascalHDXWorkbenchSoftware2012">
<p>Pascal, Bruce D., Scooter Willis, Janelle L. Lauer, Rachelle R. Landgraf, Graham M. West, David Marciano, Scott Novick, Devrishi Goswami, Michael J. Chalmers, and Patrick R. Griffin. 2012. “HDX Workbench: Software for the Analysis of H/D Exchange MS Data.” <em>Journal of the American Society for Mass Spectrometry</em> 23 (9): 1512–21. <a href="https://doi.org/10.1007/s13361-012-0419-6">https://doi.org/10.1007/s13361-012-0419-6</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#about-hadex">About HaDeX</a><ul class="nav nav-pills nav-stacked">
<li><a href="#comparison-of-the-existing-hdx-ms-software">Comparison of the existing HDX-MS software</a></li>
      </ul>
</li>
      <li>
<a href="#hadex-functionalities">HaDeX functionalities</a><ul class="nav nav-pills nav-stacked">
<li><a href="#data-import">Data import</a></li>
      <li><a href="#computation-of-deuteration-levels">Computation of deuteration levels</a></li>
      <li><a href="#difference-of-deuteration-levels-between-two-states">Difference of deuteration levels between two states</a></li>
      <li><a href="#visual-data-analysis">Visual data analysis</a></li>
      <li><a href="#woods-plot">Woods plot</a></li>
      <li><a href="#additional-tools">Additional tools</a></li>
      </ul>
</li>
      <li><a href="#hadex-graphical-user-interface">HaDeX Graphical User Interface</a></li>
      <li>
<a href="#examples">Examples</a><ul class="nav nav-pills nav-stacked">
<li><a href="#example-1-cd160-hvem">Example 1: CD160-HVEM</a></li>
      <li><a href="#example-workflow-2">Example workflow 2</a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Weronika Puchala, Michal Burdukiewicz.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9000.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
